hibernate-types-4/src/main/java/com/vladmihalcea/hibernate/type/range/guava/PostgreSQLGuavaRangeType.java

```diff
@@ -211,12 +221,11 @@ public Long apply(String s) {
     public String asString(Range range) {
         StringBuilder sb = new StringBuilder();
 
-
-        sb.append(range.lowerBoundType() == BoundType.CLOSED ? '[' : '(')
+        sb.append(range.hasLowerBound() && range.lowerBoundType() == BoundType.CLOSED ? '[' : '(')
```

Comment:
Same for this change. This fix is actually necessary for the proper working of the type, I have it applied locally as well (just hadn't the time to push it upstream yet).